modules/opensearch/opensearch.client.inc

```diff
@@ -454,19 +497,20 @@ function opensearch_execute(TingClientRequest $request) {
       }
     }
 
-    // This have to be execute after the post_execute hook as the $res and
-    // $response may not match i size and give some strange requirements in the
-    // hooks.
-    if ($request instanceof TingClientObjectRequest) {
-      // Merge in any cached objects. The map uses ids as keys. The order does
-      // not matter for object requests.
-      $response = array_merge($cached_objects, $response);
-    }
-
     // Update the cache with the raw response object from the data well for
     // faster processing. Also set the static cache.
     _opensearch_cache($request, $response);
 
+    // Merge any cached objecst that we found earlier with the result from the
```

Comment:
If this was moved above and the request pid/local was reset. It could be store in cache. Else the store partial search it not likely to be search for again?

Reply:
I get what you're saying, but I think it's a minor issue. In the next request for the same objects, the response will be reassembled from the cached objects and the full response will actually [be stored in the cache](https://github.com/vejlebib/ding2/blob/4194-opensearch-cache-local-ids/modules/opensearch/opensearch.client.inc#L473)

I think that if we should make this even better, we should instead store the individual objects in the cache. That way there's a much higher chance that a given object request will be able to re-assemble the response from cache without requesting opensearch.

Reply:
Wait.. you're right. It will actually make a new request if the $response we put in cache has more that one object. 

So we need to change something here. I can follow your suggestion and restore the original ids in the request or I can store the individual objects in cache as I suggested.

What do you think?

Reply:
I realize that we don't have much time, so I've just started working on you suggestion now. Will update PR soon.

Reply:
The individual objects will be stored in the predict cache function.

Reply:
Unfortunately only for collection and search request and not getObject request for mutliple objects :(

I Updated the PR with your suggestion to restore original request before setting cache. A nice side effects of this was that I could remove the code that converted the ids to string values, since we remember the original array.

Reply:
:-)